### PR TITLE
When common name is an ip address, add san entry as both iPAddress and dNSName

### PR DESCRIPTION
**Pull request by vipeller**
This is a workaround of a problem happened with clients based on OpenSSL. When the host name is an ip address and the server certificate has a dNSAddres as a Subject Alternative Name entry, then OpenSSL ignores iPAddress entries.
Currently OpenSSL accepts only if
- there is not dNSName (like "edgehub" in our case), in this case it check iPAddress entries and the certificate validation passess
- there is a dNSName like "edgehub", but in this case also there is an extra dNSName for the ip address of the host

because we need "edgehub" alternative name for API Proxy, we need the second workaround to make it work

### CODE REVIEW

File: edgelet/edgelet-utils/src/lib.rs
Comment (varunpuranik) [inline edgelet/edgelet-utils/src/lib.rs:84]: Was this never used anywhere?

**Review by varunpuranik — APPROVED** [review]

**Review by arsing — APPROVED** [review]

File: edgelet/edgelet-utils/src/lib.rs
Comment (vipeller) [reply edgelet/edgelet-utils/src/lib.rs:84]: it used to be used, but then after some refactor a different code took over that functionality (the one changed by this PR).

**Review by massand — APPROVED** [review]

**Review by varunpuranik — CHANGES_REQUESTED** [review]
We need to review this fix one more time, so let's keep the PR open for now. I will approve once we get ack from everyone.

**Comment by vipeller** [comment]
this was fixed a different way (through the c-sdk), so not needed

**Pull request closed without merging**
